### PR TITLE
Post Reblogging - Add wp:image markup to posts reblogged with Block Editor

### PR DESCRIPTION
**Pull request by Gio2018**
#### Release notes have been updated.

Ref #12960

To test:
wp:image markup
- Make sure you have a test blog with Block Editor enabled
- Navigate to the reader and choose a post with a featured image
- Either from post list or post detail, tap on `reblog`, and choose the site with block editor enabled
- verify that the image is displayed after the title and before the summary

Reblog button visibility
- log in with a self hosted site (Login -> Login with WordPress -> Login by entering your site address)
- Navigate to the reader, and verify that the reblog button is not visible on any post

Self hosted sites not visible in the blog selector
- Login with a user that has at least one self hosted site
- Navigate to the reader, tap the reblog button and verify that:
  - If the user has no WordPress.com sites (regardless self hosted sites), the no sites screen is presented
  - if the user has only one WordPress.com site (regardless self hosted site): 
    - the editor is presented directly (no blog selector)
    - The blog selector button does not show on the nav bar, in the editor.
  - if the user has more than one WordPress.com site:
     - the blog selector is presented, but no self hosted site is listed in the blog selector
     - when choosing the blog selector from the editor nav bar, self hosted sites are not listed in there.

PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Reader/ReaderReblogFormatter.swift [hunk 1]
```diff
@@ -27,24 +26,32 @@ struct ReaderReblogFormatter {
 // MARK: - Gutenberg formatter helpers
 private extension ReaderReblogFormatter {
 
+    static func embedInWpImage(image: String, size: String = "size-large") -> String {
+        return "<!-- wp:image {\"className\":\"\(size)\"} -->\n" + gutenbergFigure(image: image, size: size) + "\n<!-- /wp:image -->"
+    }
+
     static func embedInWpParagraph(html: String) -> String {
```
[the remark this answers is not in the document]
Comment (Gio2018) [reply WordPress/Classes/ViewRelated/Reader/ReaderReblogFormatter.swift]: Yup, removed.

File: WordPress/Classes/Services/BlogService.m [hunk 2]
```diff
@@ -510,6 +510,11 @@ - (NSArray *)blogsWithNoAccount
     return [self blogsWithPredicate:nil];
 }
 
+- (NSArray<Blog *> *)blogsForWPComAccounts
+{
+    return [self blogsWithPredicate:[NSPredicate predicateWithFormat:@"account != NULL"]];
```
Comment (ScoutHarris) [inline WordPress/Classes/Services/BlogService.m]: The web reblog doesn't show hidden sites. I don't think the app should either (pinging @frosty for confirmation). I don't see an existing method for doing this (blog list has it's own logic), so I think we could update this to be  `visibleBlogsForWPComAccounts`, and include the `predicateForVisibleBlogs`.

Comment (Gio2018) [reply WordPress/Classes/Services/BlogService.m]: Yes, even more so, the blog selector already filters the hidden sites.

**Comment by ScoutHarris** [comment]
> on an account that has no WP sites and a self-hosted site, the reblog button does not appear in the reader.

To recap a Slack convo, @Gio2018 pointed out this happens with Likes as well, and there may be an issue with that logic, pre-dating this feature. He'll follow up separately.

File: WordPress/Classes/Services/BlogService.m (hunk 2 above)
Comment (frosty) [reply WordPress/Classes/Services/BlogService.m]: I agree, we should match what the web is doing 👍

**Pull request merged**